sympy/core/power.py

```diff
@@ -1583,8 +1584,8 @@ def mul(d1, d2):
             # Convert floats like 0.5 to exact SymPy numbers like S.Half, to
             # prevent rounding errors which can induce wrong values of d leading
             # to a NotImplementedError being returned from the block below.
-            from sympy.simplify.simplify import nsimplify
-            _, d = nsimplify(g).leadterm(x, logx=logx)
+            g.replace(lambda x: x.is_Float, lambda x: Rational(x))
```

Comment:
```suggestion
            g = g.replace(lambda x: x.is_Float, lambda x: Rational(x))
```

Reply:
Really overlooked this from the start :), thanks for pointing out.